### PR TITLE
[Feature] ColorJitter and Lighting

### PR DESCRIPTION
**Pull request by LXXXXR**
Hi,
This pull request implements 2 data augmentation, including color jitter and lighting. 
Pls kindly take a look. Thank you.

### CODE REVIEW

File: mmcls/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -501,6 +500,107 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class ColorJitter(object):
```
Comment (MeowZheng) [inline mmcls/datasets/pipelines/transforms.py:504]: should we add hue adjustment?

File: mmcls/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -501,6 +500,107 @@ def __repr__(self):
...
+        contrast (float): How much to jitter contrast.
+            contrast_factor is chosen uniformly from
+            [max(0, 1 - contrast), 1 + contrast].
+        saturation (float): How much to jitter saturation.
+            saturation_factor is chosen uniformly from
+            [max(0, 1 - saturation), 1 + saturation].
+    """
+
+    def __init__(self, brightness, contrast, saturation):
+        self.brightness = brightness
```
Comment (ycxioooong) [inline mmcls/datasets/pipelines/transforms.py:520]: any constraints on these parameters?

File: mmcls/datasets/pipelines/transforms.py [hunk 3]
```diff
@@ -540,8 +640,7 @@ class Albu(object):
     """
 
     def __init__(self, transforms, keymap=None, update_pad_shape=False):
-        if Compose is None:
-            raise RuntimeError('albumentations is not installed')
+        from albumentations import Compose
```
Comment (ycxioooong) [inline mmcls/datasets/pipelines/transforms.py:644]: if `albumentations` is None, will the raised error message be clear enough?

**Review by ycxioooong — CHANGES_REQUESTED** [review]

File: mmcls/datasets/pipelines/transforms.py (hunk 2 above)
Comment (LXXXXR) [reply mmcls/datasets/pipelines/transforms.py:520]: They will be checked when calling it.

Comment (LXXXXR) [reply mmcls/datasets/pipelines/transforms.py:520]: Or do you mean I should repeat the check in `__init__` ?

**Review by ycxioooong — APPROVED** [review]

**Comment by shiyegao** [comment]
Is there a way to set the parameter 'hue', like ransforms.ColorJitter(0.4, 0.4, 0.4, 0.1)

**Comment by mzr1996** [comment]
> Is there a way to set the parameter 'hue', like ransforms.ColorJitter(0.4, 0.4, 0.4, 0.1)

The hue transform is not supported by mmcv by now, I think you can open an issue there, and then we can support this function.

**Pull request merged**